.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "tasks/computer-vision/image-generation/text-to-image/AttnGAN"]
```

Comment:
Is it possible to have seperate submodule url for each task? then propably it would be better to name it with matching task and having scope to add different sub modules later and pull only specific sub module with specific usecase? Please suggest and update accordingly.

Reply:
It is possible to have a separate submodule url for each task. It is also worth noting that a submodule is in a detached HEAD state, and is only a reference to specific commits in the submodule's repo. This means several things:
1. It is possible to have two independent submodules that come from the same repo but are of different versions.
2. Updates to a submodule repo will not be automatically pushed to the main (ai-platform) repo. These changes must be pushed manually by an ai-platform contributor, for example by using (:
```
git submodule update --remote # Updates all submodules.
```
or
```
git submodule update --remote <path to the submodule> # Update a specific submodule.
```
This syncs the submodule(s) with their repo(s) to the branch specified in `<reporoot>/.gitmodules` (default: master).


A user can also pull only the submodules that they want (https://stackoverflow.com/a/31471460/9954163):
```
git clone https://github.com/mlaradji/ai-platform.git # This shouldn't download any submodules
cd ai-platform
git submodule update --init -- tasks/computer-vision/image-generation/text-to-image/AttnGAN/ # This pulls the ai-platform version of AttnGAN
```